modal/sandbox.py

```diff
@@ -91,7 +91,45 @@ async def _get_logs():
         return data
 
 
+class _StreamWriter:
+    """Provides an interface to buffer and write logs to a sandbox stream (`stdin`)."""
+
+    def __init__(self, sandbox_id: str, client: _Client):
+        self._index = 1
+        self._sandbox_id = sandbox_id
+        self._client = client
+        self._is_closed = False
+        self._buffer = bytearray()
+
+    def get_next_index(self):
+        index = self._index
+        self._index += 1
+        return index
+
+    def write(self, data):
```

Comment:
I made `data` type-less but asserted that `isinstance(data, (bytes, bytearray, memoryview))`. This mirrors the API and semantics of `StreamWriter::write`

Reply:
Why not have a union type annotation?

Reply:
that's true I guess we're not tied to `asyncio.subprocess`'s API

Reply:
Yeah — I'd argue anyway that we have the same API, we're just enforcing it only at runtime. Adding annotations will allow us to enforce is at check time too.